modules/sdk-core/src/bitgo/wallet/iWallet.ts

```diff
@@ -485,6 +485,35 @@ export interface ShareWalletOptions {
   disableEmail?: boolean;
 }
 
+export interface BulkCreateShareOptions {
+  shareOptions: Array<{
+    user: string;
+    permissions: string;
+    keychain: {
+      pub: string;
+      encryptedPrv: string;
+      fromPubKey: string;
+      toPubKey: string;
+      path: string;
+    };
+  }>;
+}
+
+export interface BulkShareWalletOptions {
+  walletPassphrase: string;
+  shareOptions: Array<{
+    user: string;
+    permissions: string;
+    keychain: {
+      pub: string;
+      encryptedPrv?: string;
+      fromPubKey?: string;
+      toPubKey?: string;
+      path?: string;
+    };
+  }>;
+}
```

Comment:
why is this being retyped?? 
```suggestion
  shareOptions: BulkKeyShareOption[]
```

Reply:
Can you explain why they are optional in one and required in another? I see them being required in all key share operations.

Reply:
sure, 
The bulk version of this method has been kept in sync with the non bulk version implemented here
CreateShareOptions - https://github.com/BitGo/BitGoJS/blob/master/modules/sdk-core/src/bitgo/wallet/iWallet.ts#L453-L473
Moreover, In the method `bulkWalletShare`, we are generating new encryptedPrv, as well as pub and creating a `sharedKeychain`.

While in the `createBulkKeyShares` method we need all the parameters in the keychain.